src/Microsoft.Extensions.Configuration.Abstractions/IConfiguration.cs

```diff
@@ -6,12 +6,29 @@
 
 namespace Microsoft.Extensions.Configuration
 {
+    /// <summary>
+    /// Represents a set of key/value application configuration properties.
+    /// </summary>
     public interface IConfiguration
     {
+        /// <summary>
+        /// Gets or sets a configuration value.
+        /// </summary>
+        /// <param name="key">The configuration key.</param>
+        /// <returns>The configuration value.</returns>
         string this[string key] { get; set; }
 
+        /// <summary>
+        /// Gets a configuration sub-section with the specified key.
+        /// </summary>
+        /// <param name="key">The key of the configuration section.</param>
+        /// <returns>The <see cref="IConfigurationSection"/>.</returns>
         IConfigurationSection GetSection(string key);
 
+        /// <summary>
+        /// Gets all configuration sub-sections.
```

Comment:
It might be worth clarifying that these are the immediate descendant sub-sections, as opposed to all 'child' sections.  Assuming I remember correctly what this does :)